deploy-ecs/deploy.sh

```diff
@@ -15,6 +15,10 @@ set -e -u
 # - ECS_TASK_DEF
 # - DOCKER_TAG
 
+# makes 'build_register_task_options' function available, produces 'reg_options' global var
+. ../shared/register_task_options.sh
+reg_options=() # returned as global by build_register_task_options
```

Comment:
This feels a little hacky, but bash has no way to directly return array from functions

Reply:
Could you return the expanded string instead of the array?

Reply:
My understanding is you can only return a number (exit status) 0-255

Reply:
That's technically true, but since functions are treated like little commands, you can do something like:

```bash
build_register_task_options() {
  echo "--hello=world"
}

aws register-task $(register_task_options())
```

Reply:
I can't get that to work for this use case. Using echo malforms the json elements of the array by adding extra single quotes all over the place and the aws command fails

Reply:
Okay. In the interest of unblocking you, I'll approve the existing version and we can come back to this later.